server/middleware/token.js

```diff
@@ -81,10 +77,34 @@ function token(options) {
   if (typeof currentUserLiteral === 'string') {
     currentUserLiteral = escapeRegExp(currentUserLiteral);
   }
-  assert(typeof TokenModel === 'function',
-    'loopback.token() middleware requires a AccessToken model');
 
   return function(req, res, next) {
+    var app = req.app;
+    var registry = app.registry;
+    var BaseAcessToken = registry.getModel('AccessToken');
+    var TokenModel;
+
+    switch (typeof options.model) {
+      case 'string':
+        TokenModel = registry.getModel(options.model);
+      break;
+      case 'function':
+        TokenModel = options.model;
+      break;
+    }
```

Comment:
Resolving a "string or constructor" into a model constructor is IMO a common use case that should be implemented in registry, e.g. as `registry.resolveModel(nameOrCtor)`. Would you mind making this improvement as part of this PR?